pkg/app/client/client.go

```diff
@@ -243,8 +243,8 @@ type Client struct {
 	// If Proxy is nil or returns a nil *URL, no proxy is used.
 	Proxy protocol.Proxy
 
-	// RetryIfFunc sets the retry decision function. If nil, the client.DefaultRetryIf will be applied.
-	RetryIfFunc client.RetryIfFunc
+	// RetryIf sets the retry decision function. If nil, the client.DefaultRetryIf will be applied.
+	RetryIf client.RetryIf
```

Comment:
I think this PR does not seem to be a feature, please change the title and content of the PR @BaiZe1998

Reply:
i got it, thx